auth/src/main/java/org/aerogear/mobile/auth/user/UserPrincipalImpl.java

```diff
@@ -128,6 +145,20 @@ public Builder withRoles(final Set<UserRole> roles) {
             return this;
         }
 
+        public Builder withRealmRoles(final Set<UserRole> realmRoles) {
+            if (realmRoles != null) {
+                this.realmRoles.addAll(realmRoles);
+            }
+            return this;
+        }
+
+        public Builder withResourceRoles(final Set<UserRole> resourceRoles) {
```

Comment:
I don't think we need those methods. They can be retrieved from the existing `roles` set. They will have different `RoleType` in the roles set.